File: 02-tech/03-software-technologies/08-exam-preparation/03-kanban-board/CSharp-Skeleton/TeisterMask/Models/Task.cs
namespace TeisterMask.Models
{
    using System.ComponentModel.DataAnnotations;

    public class Task
    {
        [Key]
        public int Id { get; set; }

        [Required(AllowEmptyStrings = false)]
        public string Title { get; set; }

        [Required]
        public string Status { get; set; }

        public Task() { }

        public Task(string title, string status)
        {
            this.Title = title;
            this.Status = status;
        }
    }
}

File: 02-tech/03-software-technologies/08-exam-preparation/03-kanban-board/JavaScript-Skeleton/models/task.js
const Sequelize = require('sequelize');

/**
 *
 * @param {Sequelize.Sequelize} sequelize
 * @returns { Sequelize.Model }
 */
module.exports = function (sequelize)
{
    let Task = sequelize.define("Task",
    {
        title:
        {
            type: Sequelize.TEXT,
            allowNull: false,
            // @ts-ignore
            required: true,
            validate:
            {
                notEmpty: true
            }
        },
        status:
        {
            type: Sequelize.ENUM,
            values: ["Open", 'In Progress', 'Finished']
        }
    },
    {
        timestamps: false
    });

    return Task;
};